src/content/add-to-app/ios/add-flutter-screen.md

```diff
@@ -42,17 +42,19 @@ Where you create a `FlutterEngine` depends on your host app.
 {% tabs "darwin-framework" %}
 {% tab "SwiftUI" %}
 
-In this example, we create a `FlutterEngine` object inside a SwiftUI `ObservableObject`. 
-We then pass this `FlutterEngine` into a `ContentView` using the 
- `environmentObject()` property. 
+In this example, we create a `FlutterEngine` object inside a SwiftUI [`Observable`][] 
+object called `FlutterDependencies`. 
+Pre-warm the engine by calling `run()`, and then inject this object 
+into a `ContentView` using the `environment()` view modifier. 
 
  ```swift title="MyApp.swift"
```

Comment:
About half the content is the normal app boilerplate, and the other half is Flutter additions. It is easy to miss additions.

For example, the engine is injected using the `environment(...)` view modifier. This is nested in the `Scene` declaration. Forgetting to add this won’t result in an error until subsequent steps.

Should we make the code samples in this page diffs? That would make it easier which lines need to be added for Flutter.